maestrowf/datastructures/yamlspecification.py

```diff
@@ -162,21 +172,11 @@ def verify_description(self):
         # and study.
         # We're REQUIRING that user specify a name and description for the
         # study.
-        try:
-            if not self.description:
-                raise ValueError("The 'description' key is required in the "
-                                 "YAML study for user sanity. Provide a "
-                                 "description.")
-            else:
-                if not (self.description["name"] and
-                   self.description["description"]):
-                    raise ValueError("Both 'name' and 'description' must be "
-                                     "provided for a valid study description.")
-        except Exception as e:
-            logger.exception(e.args)
-            raise
 
-        logger.info("Study description verified -- \n%s", self.description)
+        # check description
+        YAMLSpecification.validate_schema("description", self.description, schemas["DESCRIPTION_SCHEMA"])
```

Comment:
I got the following exception when trying to test this branch.
```
Traceback (most recent call last):
  File "/home/frank/.virtualenvs/maestrowf/bin/maestro", line 11, in <module>
    load_entry_point('maestrowf', 'console_scripts', 'maestro')()
  File "/mnt/f/Code/Python/maestrowf/maestrowf/maestro.py", line 414, in main
    rc = args.func(args)
  File "/mnt/f/Code/Python/maestrowf/maestrowf/maestro.py", line 127, in run_study
    spec = YAMLSpecification.load_specification(args.specification)
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 112, in load_specification
    raise e
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 108, in load_specification
    specification = cls.load_specification_from_stream(data)
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 151, in load_specification_from_stream
    specification.verify()
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 157, in verify
    self.verify_description()
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 177, in verify_description
    YAMLSpecification.validate_schema("description", self.description, schemas["DESCRIPTION_SCHEMA"])
  File "/mnt/f/Code/Python/maestrowf/maestrowf/datastructures/yamlspecification.py", line 390, in validate_schema
    validator = jsonschema.Draft7Validator(schema)
AttributeError: module 'jsonschema' has no attribute 'Draft7Validator'
```

It looks like the `jsonschema` packages has a `Draft4Validator` so the `7` looks like a typo. I'll try making that change locally and see if it works.

Reply:
Oh interesting. It looks like `Draft7Validator` exists at least in `v3.2.0`, but I don't know the difference between these validators yet.

Reply:
Yeah, it looks like pip on my end defaulted to `v2.6.0` -- looks like it was already in my `site-packages` actually. Trying in a fresh environment now.

Reply:
Yup -- worked. Sorry for the goose chase. This part is resolved.
```
Collecting jsonschema
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
```